sdk/core/Azure.Core/src/Response.cs

```diff
@@ -35,5 +35,10 @@ public static Response<T> FromValue<T>(Response response, T value)
         {
             return new ValueResponse<T>(response, value);
         }
+
+        public static Response<T> FromFunc<T>(Response response, Func<T> func)
```

Comment:
Does it have to be a func? This might be a trap for people thinking that result would get cached.

Reply:
What do you recommend instead?

Reply:
FromException()?

Reply:
It just have an internal derived class that throws NoBodyException and avoid any public factory methods.